src/core/models/payloads/tagged_data.c

```diff
@@ -240,21 +241,30 @@ void tagged_data_print(tagged_data_t *tagged_data, uint8_t indentation) {
     return;
   }
 
+  printf("%sTagged Data:\n", PRINT_INDENTATION(indentation));
+
   // tag
   if (tagged_data->tag) {
     char tag[TAGGED_DATA_TAG_MAX_LENGTH_BYTES / 2] = {0};
     if (hex2string((char const *)tagged_data->tag->data, (uint8_t *)tag, TAGGED_DATA_TAG_MAX_LENGTH_BYTES / 2) == 0) {
-      printf("%sTag: %s\n", PRINT_INDENTATION(indentation), tag);
+      printf("%s\tTag: %s\n", PRINT_INDENTATION(indentation), tag);
     }
   } else {
-    printf("%sTag:\n", PRINT_INDENTATION(indentation));
+    printf("%s\tTag:\n", PRINT_INDENTATION(indentation));
   }
 
   // binary data
   if (tagged_data->data) {
-    printf("%sData: ", PRINT_INDENTATION(indentation));
-    dump_hex_str(tagged_data->data->data, tagged_data->data->len);
```

Comment:
what if data is binary, not string? we should represent the original data in hex string, the user can decode as needed.